ambry-tools/src/main/java/com.github.ambry/store/BlobStatus.java

```diff
@@ -13,44 +13,42 @@
  */
 package com.github.ambry.store;
 
-import java.util.ArrayList;
+import com.github.ambry.utils.Utils;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 
 /**
  * Holds status of a blob from the perspective of an Index. If multiple index entries are found for the same blob,
  * everything is captured in a single instance of this class
  */
 class BlobStatus {
-  private final Set<String> available;
-  private final Set<String> deletedOrExpired;
-  private final Set<String> unavailable;
+  private final Set<String> available = new HashSet<>();
+  private final Set<String> deletedOrExpired = new HashSet<>();
+  private final Set<String> unavailable = new HashSet<>();
+  private long earliestPutTimeMs = Utils.Infinite_Time;
+  private long earliestDeleteTimeMs = Utils.Infinite_Time;
   private boolean isDeletedOrExpired;
+  private AtomicBoolean belongsToRecentIndexSegment = new AtomicBoolean(false);
 
-  BlobStatus(String replica, boolean isDeletedOrExpired, ArrayList<String> replicaList) {
-    available = new HashSet<>();
-    deletedOrExpired = new HashSet<>();
-    unavailable = new HashSet<>();
-    this.isDeletedOrExpired = isDeletedOrExpired;
-    if (!isDeletedOrExpired) {
-      available.add(replica);
-    } else {
-      deletedOrExpired.add(replica);
-    }
+  BlobStatus(List<String> replicaList) {
     if (replicaList != null && replicaList.size() > 0) {
       unavailable.addAll(replicaList);
-      unavailable.remove(replica);
     }
   }
 
   Set<String> getAvailable() {
```

Comment:
this is called `getAvailable()` but the one that returns `unavailable` is called `getUnavailableList()`. Would prefer if naming is consistent.